### DISCUSSION TITLE
Improving inference for the mouse pose

### DISCUSSION THREAD

**Discussion by singlakdeepak** [talmolab/sleap#1675] [Help!]
Hello,  

I am trying to generate a sleap model for a single mouse in the frame. I am currently using SLEAP 1.3.3  and train a "multi-animal top-down" model with max instances =1 and "tail" as the anchor point. Please see a screenshot of the points tracked. 
![mouse_frame](https://github.com/talmolab/sleap/assets/22357669/52cf26ed-1986-4912-a118-1fd698f68cb6)
However, even after labeling more than 200 frames in different poses, the left paws and right paws are predicted as jumbled i.e. left paws are predicted as right paws and vice versa in many of the frames. Of course, there are some frames where predictions come out correctly but it happens consistently a lot of times. Please see one of the erroneous predicted frame here. 
![error_image](https://github.com/talmolab/sleap/assets/22357669/4600ad8e-6ca4-4be4-9e4c-fecbd98e2703)

The videos that I use for tracking are generally 30 minute long (recorded at 80 fps) with framesize of 2048 x 2048.
Also, please see the pics of the model configuration attached. 
![train1_img](https://github.com/talmolab/sleap/assets/22357669/740699d2-7a91-4bab-9066-c59886989531)
![train2_img](https://github.com/talmolab/sleap/assets/22357669/02914491-1275-410a-a4ab-c8cb981d629d)
![train3_img](https://github.com/talmolab/sleap/assets/22357669/2370f051-2f4f-466f-8f48-fc2ac345029b)

I would really appreciate if someone can help me out here and help with predicting better. Please let me know if I need to change the model configurations to something else. Thank you!

**Comment by Marielamend** [comment]
Hello @singlakdeepak!

Thank you for all the screenshots + info you provided!

Most likely your issue stems from labeling, you can go through your existing labels again and check for this inconsistency.
We often see this happen when there are 1-2 frames in which the right/left nodes were labeled incorrectly. This can cause issues with the model leading to these constant switches. 


Let us know if this helps and if you have any other issues!
Mariela

**Reply by singlakdeepak** [reply]
Hello Mariela,

Thank you for your reply. I went through each labeled frame and it doesn't seem there was any inconsistency in the labeling. I further added/corrected about 60-70 more frames where this error happened and trained a new model. However, I am still finding plenty of frames where the nodes are incorrectly predicted. Could it happen because of different anchor points? Or any other training configurations.

**Reply by Marielamend** [reply]
Hello @singlakdeepak,

You can try to set the max stride to 64 to increase the receptive field size. The receptive field is represented by a blue box on the left of the model configuration window. This will encourage the model to learn more global context that will help distinguish between the right and left nodes. 

Let us know if that works! 
-Mariela
